addon/components/inputs/autocomplete.js

```diff
@@ -16,10 +16,29 @@ export default SelectInput.extend({
   getDefaultProps () {
     return {
       ignoreEmptyFilterSearch: true,
-      filter: ''
+      filter: '',
+      _focused: false,
+      _isTyping: false
     }
   },
 
+  // == Observers ====================================================
+
+  observeSelectedItemLavelChange: observer('value', 'options.[]', '_isTyping', 'filter', function () {
+    const {
+      _isTyping: isTyping,
+      filter,
+      options,
+      value
+    } = this.getProperties('value', 'options', '_isTyping', 'filter')
+    const selectedItem = this._findSelectedItemGivenValue(value, options)
+    if (!isEmpty(selectedItem)) {
+      const label = get(selectedItem, 'label')
+      if (!isEmpty(label) && isEmpty(filter) && !isTyping) this.set('filter', label)
```

Comment:
So I'm having a tough time understanding what the fix is. Do you mind posting screenshots/vids of the before and after. From what I can understand from these changes, it looks like the `filter` was empty previously and you're setting it the the selected item label when user is not typing? Is that correct?

Reply:
Well the original problem when you have a selectedItem and backspace to an empty filter it would.
![2018-07-19 11_13_08](https://user-images.githubusercontent.com/9057680/42951681-c3ad404e-8b44-11e8-9555-0ee947e3af63.gif)

So if I can keep track of typing I know that I shouldn't set the filter when the user is just backspacing all the way
![2018-07-19 11_01_52](https://user-images.githubusercontent.com/9057680/42952114-b9798b2c-8b45-11e8-8ef1-e4d0b7c850b7.gif)

Reply:
Interesting. Originally I was thinking that if a user started typing, it meant user intention was to set the value (even with backspace to clear) so it shouldn't need to set the filter back to the original. But `autocomplete` isn't like Google's auto suggest. A value is only made with a selection and filtering the list shouldn't clear the value.

This was easier to handle in `select` because there was clear separation between the filtering input and the value's input element. In `select` typing anything would update the filter to whatever the user typed, regardless if it was empty. If the user didn't like the choices, he could close the dropdown and get back his original value.

Would it be better to keep track of the dropdown states instead of the typing? If the dropdown is open, don't override the filter ever; just keep it at whatever the user typed. Hmm. that would mean you'd also have to fix the automatic closing of the dropdown when they backspace to clear. Man, this component is getting really complex.

Up to you I guess. If I had a hard time understanding this scenario, others will and it'll be hard to maintain. Perhaps throw in some comments explaining this.

Reply:
The problem, in this case, is I don't have knowledge on whether drop-down is open or not. That's in `frost-autocomplete`, and since we need to use `onInput` for querying/etc we can't really have `frost-autocomplete` control filter for us (because `frost-autocomplete` assumes if you use `onInput` you will control the `filter`, and shouldn't set it.


I think I made it much more readable now by removing the observer logic, I'll add some comments on `_isTyping` to make it easier to read

Reply:
Looks like drop-down state is also closed when you have an empty filter. So I'd run into the same issue

Reply:
Thanks, removing the observer business definitely makes this more readable.